src/com/facebook/buck/jvm/java/PrebuiltJar.java

```diff
@@ -275,15 +275,7 @@ public boolean neverMarkAsUnusedDependency() {
 
   @Override
   public Stream<BuildTarget> getRuntimeDeps(BuildRuleResolver buildRuleResolver) {
-    Stream<BuildTarget> transitiveRuntimeDeps = Stream.of();
-    for (JavaLibrary lib : getTransitiveClasspathDeps()) {
-      // Skip ourself to avoid infinite recursion.
-      if (lib == this) continue;
-
-      transitiveRuntimeDeps =
-          Stream.concat(transitiveRuntimeDeps, lib.getRuntimeDeps(buildRuleResolver));
-    }
-    return transitiveRuntimeDeps;
+    return Stream.of();
```

Comment:
I don't know that we can do this.
I doubt the PrebuiltJars are only ever dependencies of JavaBinary, so this will be wrong for all the other places that uses it.

Reply:
This change mirrors what `DefaultJavaLibrary` does - it does not attempt to get the runtime deps of its' deps: https://github.com/facebook/buck/blob/master/src/com/facebook/buck/jvm/java/DefaultJavaLibrary.java#L436 (it's kind of a journey to trace through, but this is where there `Description` gets turned into that runtimeDeps list: https://github.com/facebook/buck/blob/master/src/com/facebook/buck/jvm/java/JavaLibraryDeps.java#L127). Both `prebuilt_jar` and `java_library` are noops when built standalone.

Presumably all other users (including `java_binary`) want to support both 3rd-party (`prebuilt_jar`) and locally-built (`java_library`) - in which case they must do the transitive closure themselves. Anyone who depended on `prebuilt_jar` hoisting runtime deps was likely already broken. The original PR that added these (https://github.com/facebook/buck/commit/b22d68aba5a046a91bac38363a5695e72afdb459) didn't touch anything other than `java_binary` though there's a chance more support has been added in the past 2 months.